src/routes/index.ts

```diff
@@ -138,17 +138,28 @@ const DefaultRouteRequest = z.object({
 });
 
 const defaultRouteHandler: RequestHandler<{}, {}, z.infer<typeof DefaultRouteRequest>> = async (req, res, next) => {
-  const { step } = DefaultRouteRequest.parse(req.body);
-  if (!step) {
-    throw new Error('no step');
+  try {
+    const { step } = DefaultRouteRequest.parse(req.body);
+    if (!step) {
+      throw new Error('no step');
+    }
+    const handler = Object.prototype.hasOwnProperty.call(steps, step) && steps[step];
+    if (!handler) {
+      throw new Error(`invalid step ${step}`);
+    }
+  
+    await handler(req, res, next);
   }
-
-  const handler = Object.prototype.hasOwnProperty.call(steps, step) && steps[step];
-  if (!handler) {
-    throw new Error(`invalid step ${step}`);
+  catch (e) {
+    if (e instanceof Error) {
```

Comment:
what else might e be an instance of here?

Reply:
Oh yep - I don't need this line